### PR TITLE
AppSecrets section and first row completion

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

The goal of this PR is to include a new component that shows the secrets of the application in a more prominent place, along with some style changes and internal to finish the "first" row of the application view. The goal of this first row is to try to show everything needed to access the repo: readiness, access URL and credentials:

Screenshot from 2020-07-22 12-10-41

### Applicable issues

  - ref #1354

### CODE REVIEW

File: dashboard/src/shared/specs/mountWrapper.tsx [hunk 1]
```diff
@@ -7,15 +7,17 @@ import thunk from "redux-thunk";
 
 import { IStoreState } from "../../shared/types";
 
-const mockStore = configureMockStore([thunk]);
+export const mockStore = configureMockStore([thunk]);
 
-const initialState = {
+export const initialState = {
```
Comment (absoludity) [inline dashboard/src/shared/specs/mountWrapper.tsx:12]: I don't think we need these extra exports, if we use `getStore(objectToMergeWithInitialState)` that I landed earlier.  See what you think.

File: dashboard/src/shared/utils.ts [hunk 2]
```diff
@@ -33,3 +36,28 @@ export function trimDescription(desc: string): string {
...
+  resources: { [s: string]: IKubeItem<IResource | IK8sList<IResource, {}>> },
+) {
+  const result: Array<IKubeItem<IResource | ISecret>> = [];
+  refs.forEach(ref => {
+    const kubeItem = resources[ref.getResourceURL()];
+    if (kubeItem) {
+      const itemList = kubeItem.item as IK8sList<IResource | ISecret, {}>;
+      if (itemList && itemList.items) {
+        itemList.items.forEach(i => {
+          result.push({
```
Comment (absoludity) [inline dashboard/src/shared/utils.ts:51]: So this won't handle a list of lists I don't think? But maybe that's never going to be a thing (don't know of a situation in k8s of such a list anyway).

**Comment by absoludity** [comment]
> The goal of this first row is to try to show everything needed to access the repo: readiness, access URL and credentials:

Not all the credentials are actually needed, I don't think? For eg., in the example screenshot, it's only the WP secret which is relevant and needed. I wonder if it'd be worth differentiating.

**Comment by andresmgot** [comment]
> > The goal of this first row is to try to show everything needed to access the repo: readiness, access URL and credentials:
> 
> Not all the credentials are actually needed, I don't think? For eg., in the example screenshot, it's only the WP secret which is relevant and needed. I wonder if it'd be worth differentiating.

Yes, they are not. In other cases it will be even less relevant for secrets like encoded certs but we have no way of knowing which of them are "important" to build that list.

File: dashboard/src/shared/utils.ts (hunk 2 above)
Comment (andresmgot) [reply dashboard/src/shared/utils.ts:51]: I don't know if a list of lists is allowed by the K8s API, certainly I've never seen one.

**Pull request merged**